src/App.tsx

```diff
@@ -15,48 +15,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetical = 'Alphabetical',
+  Length = 'Length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortType, setSortType] = useState<SortType | null>(null);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
+
+    if (isReverse) {
+      setGoods([...sortedGoods].reverse());
+    } else {
+      setGoods(sortedGoods);
+    }
+
+    setSortType(SortType.Alphabetical);
+  };
+
+  const sortByLength = () => {
+    if (!(sortType === SortType.Length)) {
+      const sortedGoods = [...goodsFromServer].sort((a, b) => {
+        const lengthComparison = a.length - b.length;
+
+        if (lengthComparison === 0) {
+          return 1;
+        }
+
+        return lengthComparison;
+      });
+
+      if (isReverse) {
+        setGoods([...sortedGoods].reverse());
+      } else {
+        setGoods(sortedGoods);
+      }
+
+      setSortType(SortType.Length);
+    }
+  };
+
```

Comment:
try to combine these functions in one separate function that will return the sorted array of todos

it will improve at least the readability